executor/analyze.go

```diff
@@ -897,8 +904,34 @@ func (e *AnalyzeFastExec) handleSampTasks(bo *tikv.Backoffer, workID int, err *e
 }
 
 func (e *AnalyzeFastExec) buildHist(ID int64, collector *statistics.SampleCollector, tp *types.FieldType) (*statistics.Histogram, error) {
-	// TODO: build histogram and cmsketch here for one collector.
-	return nil, nil
+	// build collector properties.
+	collector.Samples = collector.Samples[:e.sampCursor]
+	err := collector.CalcTotalSize()
+	if err != nil {
+		return nil, err
+	}
+	collector.Count = int64(e.sampCursor)
```

Comment:
`e.sampCursor` may contain null rows so we cannot use it as `Count`?

Reply:
Oh, this line is useless. In fast analyze, the row count is calculated especially instead of using `collector.Count` in `BuildColumnHist`.